pkg/controller/nodenetworkconfigurationpolicy/enactmentconditions/conditions.go

```diff
@@ -94,7 +103,17 @@ func (ec *EnactmentConditions) updateEnactmentConditions(
 		if err != nil {
 			return err
 		}
-		return nil
+
+		// Wait until enactment has being updated at the node
+		expectedStatus := instance.Status
+		return wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+			err = ec.client.Get(context.TODO(), ec.enactmentKey, instance)
+			if err != nil {
+				return false, err
+			}
+
+			return reflect.DeepEqual(expectedStatus, instance.Status), nil
```

Comment:
I'm ok with silently failing here, but can we at least log the incident to debug log?